util2.py

```diff
@@ -0,0 +1,17 @@
+def parseBoolValue(value: str | None, fail_on_errors: bool = True, preserve_none: bool = False) -> bool | None:
+    """Parses a string representing a boolean value. If parsing was successful,
+       returns `True` or `False`. If `preserve_none=True`, returns `True`, `False`,
+       or `None`. If parsing was not successful, raises `ValueError`, or, if
+       `fail_on_errors=False`, returns `None`."""
+    if not isinstance(value, str):
+        if preserve_none and value is None:
+            return value
+        return bool(value)
+    elif preserve_none and value.lower() == 'none':
+        return None
+    elif value.lower() in ('true', 'yes', 'y', 'on', '1'):
+        return True
+    elif value.lower() in ('false', 'no', 'n', 'off', '0', 'none'):
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Potential inconsistency in 'none' value handling</b></div>
<div id="fix">

Consider removing `'none'` from the list of false values, as it's already handled separately for `preserve_none=True`. This might lead to unexpected behavior when `preserve_none` is `True`.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,1 +14,1 @@
 -    elif value.lower() in ('false', 'no', 'n', 'off', '0', 'none'):
 +    elif value.lower() in ('false', 'no', 'n', 'off', '0'):
```

```suggestion
    elif value.lower() in ('false', 'no', 'n', 'off', '0'):
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2482299093>#40425a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged